src/scene/shader-lib/utils.js

```diff
@@ -40,9 +42,10 @@ function createShader(device, vsName, fsName, useTransformFeedback = false) {
  * attribute names to semantics SEMANTIC_*. This enables the engine to match vertex buffer data as
  * inputs to the shader. Defaults to undefined, which generates the default attributes.
  * @param {boolean} [useTransformFeedback] - Whether to use transform feedback. Defaults to false.
+ * @param {string} [fragmentOutputType] - Output fragment shader type. Defaults to vec4.
  * @returns {Shader} The newly created shader.
  */
-function createShaderFromCode(device, vsCode, fsCode, uniqueName, attributes, useTransformFeedback = false) {
+function createShaderFromCode(device, vsCode, fsCode, uniqueName, attributes, useTransformFeedback = false, fragmentOutputType = 'vec4') {
```

Comment:
- I think its time we add `options` parameter here instead of just another parameter now that we need to add a new one. 
- I think instead of a single output type, we need an array of those, allowing us to specify different output per render target in MRT scenario
- other parts of the code need to be adjusted to handle this, but that looks straightforward - I like the way you handle it.